core/processor/abci.go

```diff
@@ -722,7 +722,7 @@ func (app *App) LoadSnapshotChunk(req tmtypes.RequestLoadSnapshotChunk) tmtypes.
 
 func (app *App) OnInitChain(req tmtypes.RequestInitChain) tmtypes.ResponseInitChain {
 	app.log.Debug("ABCI service InitChain start")
-	hash := hex.EncodeToString(vgcrypto.Hash(req.AppStateBytes))
+	hash := hex.EncodeToString(vgcrypto.Hash([]byte(req.ChainId)))
```

Comment:
The appstate bytes were coming out different on different runs, which made the "block-hash" for the genesis in the vents different. I didn't dig too deeply but I'm guessing its because tendermint parses the genesis file it puts them into maps, which it then serialises and means the bytes can be slightly different.

Reply:
hum interesting.